portal-bridge/src/bridge/era1.rs

```diff
@@ -101,12 +101,11 @@ impl Era1Bridge {
     }
 
     pub async fn launch_single(&self, epoch: u64) {
-        let epoch_string = format!("{epoch}-");
-        let era1_path = self
-            .era1_files
-            .clone()
-            .into_iter()
-            .find(|file| file.contains(&epoch_string));
+        let era1_path = self.era1_files.clone().into_iter().find(|file| {
+            file.contains("mainnet-")
+                && file.contains(&format!("{epoch}-"))
+                && file.contains(".era1")
```

Comment:
This still doesn't fix the problem. For an epoch value of `1`, this case will be true for all files
- `mainnet-00001-hash.era1`
- `mainnet-00011-hash.era1`
- `mainnet-00101-hash.era`
- etc.

You need to left-pad the epoch value with up to 5 zeroes.

Reply:
I just fixed that. but the example you gave isn't true
It would only return true for ``mainnet-00001-hash.era1``

Reply:
`{epoch}-` this dash ties it to check the right

Reply:
yeah, but it's the left side of the epoch value that's the problem

Reply:
![image](https://github.com/ethereum/trin/assets/31669092/899f31c1-93e8-4ee2-a922-b6709c311ffd)

here is proof

Reply:
I just pushed a change which will make it more readable. But the code I had worked.
![image](https://github.com/ethereum/trin/assets/31669092/8dc78a42-f6e3-4724-a0dc-3750b92460e5)

Reply:
- `mainnet-00001-hash.era1`
- `mainnet-00011-hash.era1`
- `mainnet-00101-hash.era1`

All of those strings "contain" the following substrings: `"mainnet"`, `"1-"`, and `".era1"`

Reply:
Oh I see what you mean now. I fixed it which is good. My bad for the confusion xD.

Reply:
My code did work assuming the list was sorted, but yeah the issue is the list wouldn't be sorted because it is shuffle. Sorry about that